src/sentry/api/endpoints/project_symbolsources.py

```diff
@@ -0,0 +1,76 @@
+"""API to handle symbol sources.
+
+The symbol sources are stored in a project option as JSON array of objects and can be
+directly manipulated via the project details.  This API aims to provide an easier API to
+handle changing symbol sources however.
+
+Provides the following endpoints:
+
+1. ``GET /{org_slug}/{proj_slug}/symbolsources/`` to retrieve all symbol sources as a JSON
+   array of config objects.
+
+2. ``POST /{org_slug}/{proj_slug}/symbolsources/`` to create a single new config object.
+
+
+3. ``PUT /{org_slug}/{proj_slug}/symbolsources/{id}`` to update an existing config object.
+
+4. ``DELETE /{org_slug}/{proj_slug}/symbolsources/{id}`` to delete an existing config object.
+"""
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases.project import ProjectEndpoint, StrictProjectPermission
+from sentry.models import Project
+
+
+class ProjectSymbolSourcesEndpoint(ProjectEndpoint):  # type: ignore
```

Comment:
Nit: A common pattern throughout our api is that we call the endpoint that operates on a set of data the `index` endpoint and the specific endpoint the `detail` endpiont, and keep these in separate files.

So we'd call this `ProjectSymbolSourcesIndexEndpoint` and put in `src/sentry/api/endpoints/project_symbol_sources_index.py`
and the other `ProjectSymbolSourcesDetailEndpoint` and put in `src/sentry/api/endpoints/project_symbol_sources_details.py`

Reply:
❤️  Thanks for the pointer!  I've updated this.